src/paint.rs

```diff
@@ -144,6 +174,10 @@ impl<'a> Painter<'a> {
                 {
                     line.truncate(line.len() - ANSI_SGR_RESET.len());
                 }
+
+                if config.show_line_numbers {
+                    output_buffer.push_str(&line_numbers);
+                }
```

Comment:
I have a refactoring suggestion. What do you think about deferring the formatting of the line numbers until here? You could move `prepare_line_numbers` into this module (could we call it `format_line_numbers`?), and change the code in `delta.rs` so that it accumulates the line numbers as (optional) ints rather than as formatted text.